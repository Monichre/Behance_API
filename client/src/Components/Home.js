import React, { Component } from 'react'
import Axios from 'axios'
import { Constants } from '../constants.js'

export default class Home extends Component {
    constructor(props) {
        super(props)

        this.state = {
            searchActive: false,
            searchTerm: ''
        }
    }
    componentWillMount() {
        Axios.get('/gallery').then((response) => {
            console.log(response)
        }).catch((error) => {
            console.log(error)
        })
    }
    runSearch(e) {
        e.preventDefault()
        this.setState({
            searchActive: true
        })
    }
    closeSearch(e) {
        e.preventDefault()
        this.setState({
            searchActive: false
        })
    }
    handleSearchInput(e) {
        e.preventDefault()
    }

    render() {
        return (
            <div id="Home">
                <header>
                    <i className="fa fa-bars"></i>
                    <div className="menu">
                        <div className={`search-wrapper ${this.state.searchActive ? 'active' : ''}`}>
                            <div className="input-holder">
                                <input type="text" className="search-input" value={this.state.searchTerm} onChange={this.handleSearchInput.bind(this)} placeholder="Type to search" />
                                <button className="search-icon" onClick={this.runSearch.bind(this)}><span></span></button>
                            </div>
                            <span className="close" onClick={this.closeSearch.bind(this)}></span>
                        </div>
                    </div>
                </header>

                <aside>
                    <h1>Dealer Inspire + Behance</h1>
                    <ul>

                    </ul>
                </aside>

                <div className="dashboard">
                    <canvas id="Chart"></canvas>
                    <div className="row">
                    </div>
                </div>
            </div>
        )
    }

}