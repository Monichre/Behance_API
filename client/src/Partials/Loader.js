import React from 'react'
export const Loader = () => (
    <div className="loader_bg">
        <div className="loader">
            <div className="loader__figure"></div>
            <p className="loader__label">Loading...</p>
        </div>
    </div>
)