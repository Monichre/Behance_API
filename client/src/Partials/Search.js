export const Search = (props) => (
    <div className='search-content'>
        <div className='results-column'>
            {
                this.props.searchResults.map((result) => (
                    <div className='searchResult'>
                        <h1>search result</h1>
                    </div>
                ))
            }
        </div>
    </div>
)