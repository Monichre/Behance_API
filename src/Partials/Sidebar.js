export const SideBar = () => (
    <div class="dashboard-sidebar">
        <div class="dashboard-sidebar__brand">

        </div>
        <div class="dashboard-nav">
        <ul>
            <li class="dashboard-nav__item"><a href="home"></a></li>
            <li class="dashboard-nav__item dashboard-nav__item--selected"></li>
            <li class="dashboard-nav__item"><a href="discover_items"></a></li>
            <li class="dashboard-nav__item"><a href="notifications"></a></li>
            <li class="dashboard-nav__item"><a href="settings"></a></li>
        </ul>
        </div>
</div>
)