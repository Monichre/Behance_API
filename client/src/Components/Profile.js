import React, { Component } from 'react'
import { Card, Icon, Image, List, Divider } from 'semantic-ui-react'
import Axios from 'axios'
import {Loader} from '../Partials/Loader'

export default class Profile extends Component {
    constructor(props) {
        super(props)

        this.state = {
            first_name: '',
            last_name: '',
            fields: [],
            images: [],
            occupation: '',
            location: '',
            social_links: [],
            stats: {},
            website: '',
            projects: [],
            sections: [],
            active_project: null,
            data_ready: false
        }


    }

    componentWillMount() {

        const user_id = this.props.match.params.user_id
        const _this = this

        Axios.get('/get-user', {
            params: {
                user_id: user_id
            }
        }).then((response) => {

            const { first_name, last_name, fields, images, sections, occupation, location, social_links, stats, website } = response.data.user_data
            const { projects, work_experience } = response.data
            let { followers, following } = response.data

            followers = followers.splice(0, 4)
            following = following.splice(0, 4)


            _this.setState({
                first_name: first_name,
                last_name: last_name,
                fields: fields,
                images: images,
                occupation: occupation,
                location: location,
                social_links: social_links,
                stats: stats,
                website: website,
                projects: projects,
                followers: followers,
                following: following,
                sections: sections,
                work_experience: work_experience,
                active_project: projects[0].id
            })

        }).catch((error) => {
            console.log(error)
        })
    }

    setActiveProject(project) {
        console.log(project)
        this.setState({
            active_project: project
        })
    }
    renderBackgroundImage() {
        return this.state.projects.find((project) => project.id === this.state.active_project)

    }
    componentDidMount() {
        setTimeout(() => {
            this.setState({
                data_ready: true
            })
        })
    }

    render() {

        let image_bg
        if (this.state.active_project) {
            let img_src = this.renderBackgroundImage()
            image_bg = <Image src={img_src.covers['230']} />
        }
    


        if (this.state.data_ready) {
            return (
                <div className='profile_container'>
                    <div className="profile">
                        <div className="profile-sidebar">
                            <div className="profile-sidebar__brand">
                                <Image circular='true' size='mini' src={this.state.images['50']} />
                                <br />
                                {this.state.first_name} {this.state.last_name}
                                <br />
                                <b>{this.state.occupation}</b>
                                <br />
                                {this.state.website}
                            </div>
                            <div className="profile-nav">
                                <h1>Projects</h1>
                                <ul>
                                    {this.state.projects.map((project, i) => (
                                        <li key={i} className={`profile-nav__item ${project.id === this.state.active_project ? 'profile-nav__item--selected' : ''}`}><a onClick={this.setActiveProject.bind(this, project.id)}>{project.name}</a></li>
                                    ))}
                                </ul>
                            </div>
                        </div>
                        <div className="profile-content">
                            <div className="profile-header">
                                <h2>{this.state.fields.map((field) => (<span> {field} </span>))}</h2>
                                <span>{this.state.sections.About}</span>
                            </div>
                            {this.state.projects.map((project, i) => (
                                <div key={i} className={`profile-content__panel ${project.id === this.state.active_project ? 'profile-content__panel--active' : ''}`}>
                                    <div className="profile-list">
                                        <div className="profile-list__item profile-list__item--active">
                                            <h2><img src={project.features ? project.features[0].site.ribbon.image : null} /> {project.name}</h2>
                                            <List horizontal className='profile_social_links'>
                                                {this.state.social_links.map((link) => (
                                                    <List.Item className='stat'><span><a href={link.url}>{link.service_name}</a></span> </List.Item>
                                                ))}
                                            </List>
                                            <List horizontal className='fields'>
                                                {project.fields.map((field) => (<List.Item className='stat'><span>{field}</span> </List.Item>))}
                                            </List>
                                  
                                            <div>
                                                <a href={project.url}>
                                                    <Image src={project.covers['230']} />
                                                </a>
                                            </div>

                                        </div>
                                        <List horizontal className="stats_list">
                                                {Object.keys(this.state.stats).splice(0, 4).map((stat) => (
                                                    <List.Item className='stat'> <span>{stat}:</span> {this.state.stats[stat]} </List.Item>
                                                ))}
                                            </List>
                                    </div>
                                </div>
                            ))}

                        </div>
                        <div className="profile-preview">
                            <div className="profile-preview__panel profile-preview__panel--active" style={{ backgroundImage: `url(${this.state.images['138']})`, backgroundRepeat: 'no-repeat', backgroundPosition: 'top right' }}>
                                <div className="profile-preview__header">
                                    <div className="profile-nav">
                                        <div>
                                            <h1>Following</h1>
                                            <List className='following'>
                                                {this.state.following ? this.state.following.map(followed =>
                                                    <List.Item >
                                                        <Image src={followed.images['50']} circular={true} size='mini' avatar={true} verticalAlign='top' /> <span>{followed.first_name} {followed.first_name}</span>
                                                    </List.Item>)

                                                    : null}
                                            </List>
                                        </div>
                                        <div>
                                            <h1>Followers</h1>
                                            <List className='followers'>
                                                {this.state.followers ? this.state.followers.map(follower =>
                                                    <List.Item className='stat'>
                                                        <Image src={follower.images['50']} circular={true} size='mini' avatar={true} verticalAlign='top' /> <span>{follower.first_name} {follower.first_name}</span>
                                                    </List.Item>)

                                                    : null}
                                            </List>
                                        </div>
                                    </div>
                                </div>
                                <Divider />
                                <div className="profile-preview__content work_experience">
                                    <section>
                                        <h2>Work Experience</h2>
                                        <List>
                                            {(this.state.work_experience && this.state.work_experience.length > 0) ? this.state.work_experience.map((work_item) => (
                                                <List.Item className='stat'> <b>{work_item.position}</b> at <b>{work_item.organization},</b> {work_item.location} <i>{work_item.start_date}-{work_item.end_date}</i> </List.Item>
                                            )) :

                                                null

                                            }
                                        </List>
                                    </section>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            )
        } else {
            return <Loader />
        }
    }
}