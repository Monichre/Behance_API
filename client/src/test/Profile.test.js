import React from 'react'
import sinon from 'sinon'
import Home from '../Components/Home'
import Profile from '../Components/Profile'
import { Search } from '../Partials/Search'
import { BrowserRouter, Route } from 'react-router-dom'
import runSearch from './__mocks__/search-api.js'
import { shallowToJson } from 'enzyme-to-json'
import { string, object } from 'prop-types'


describe('<Profile />', () => {
    let match = {
        params: {
            user_id: 1853107
        }
    }
    const profile_wrapper = shallow(<Profile match={match} />)
    profile_wrapper.propTypes = {
        match: object.isRequired
    }
    it('search results link to proper page and mounts with user_id', () => {

        expect(shallowToJson(profile_wrapper)).toMatchSnapshot()

    });
 
    it('sets state properly', async () => {

        let data = await runSearch()
        data[0].data_ready = true
        data[0].projects = [{
            id: 1,
            name: 'project',
            fields: ['field'],
            covers: {
                230: 'img'
            }

        }]
        data[0].sections = {
            About: 'About me'
        }
        data[0].social_links = []
        data[0].followers = []
        data[0].following = []
        data[0].work_experience = []


        const { first_name, last_name, fields, images, occupation, location, social_links, followers, following, work_experince, stats, website, projects, sections, active_project, data_ready } = data[0]

        profile_wrapper.setState({
            first_name: first_name,
            last_name: last_name,
            fields: fields,
            images: images,
            occupation: occupation,
            location: location,
            social_links: social_links,
            stats: stats,
            followers: followers, 
            following: following, 
            work_experince: work_experince,
            website: website,
            projects: projects,
            sections: sections,
            active_project: active_project,
            data_ready: data_ready

        })

        Object.keys(profile_wrapper.state()).forEach((item) => {
            expect(item).toBeTruthy()
        })
        expect(shallowToJson(profile_wrapper)).toMatchSnapshot()
        

    });


});