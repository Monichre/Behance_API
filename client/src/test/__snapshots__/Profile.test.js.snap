// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> search results link to proper page and mounts with user_id 1`] = `<Loader />`;

exports[`<Profile /> sets state properly 1`] = `
<div
  className="profile_container"
>
  <div
    className="profile"
  >
    <div
      className="profile-sidebar"
    >
      <div
        className="profile-sidebar__brand"
      >
        <Image
          as="img"
          circular="true"
          size="mini"
          src="https://mir-s3-cdn-cf.behance.net/user/50/4609491853107.58b2df5a7fef4.jpg"
          ui={true}
        />
        <br />
        Liam
         
        Pitchford
        <br />
        <b>
          Digital Artist
        </b>
        <br />
        http://www.liampitchford.com
      </div>
      <div
        className="profile-nav"
      >
        <h1>
          Projects
        </h1>
        <ul>
          <li
            className="profile-nav__item "
            key="0"
          >
            <a
              onClick={[Function]}
            >
              project
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      className="profile-content"
    >
      <div
        className="profile-header"
      >
        <h2>
          <span>
             
            Art Direction
             
          </span>
          <span>
             
            Motion Graphics
             
          </span>
          <span>
             
            Digital Art
             
          </span>
        </h2>
        <span>
          About me
        </span>
      </div>
      <div
        className="profile-content__panel "
        key="0"
      >
        <div
          className="profile-list"
        >
          <div
            className="profile-list__item profile-list__item--active"
          >
            <h2>
              <img
                src={null}
              />
               
              project
            </h2>
            <List
              className="profile_social_links"
              horizontal={true}
            />
            <List
              className="fields"
              horizontal={true}
            >
              <ListItem
                className="stat"
              >
                <span>
                  field
                </span>
                 
              </ListItem>
            </List>
            <div>
              <a>
                <Image
                  as="img"
                  src="img"
                  ui={true}
                />
              </a>
            </div>
          </div>
          <List
            className="stats_list"
            horizontal={true}
          >
            <ListItem
              className="stat"
            >
               
              <span>
                followers
                :
              </span>
               
              1251
               
            </ListItem>
            <ListItem
              className="stat"
            >
               
              <span>
                following
                :
              </span>
               
              303
               
            </ListItem>
            <ListItem
              className="stat"
            >
               
              <span>
                appreciations
                :
              </span>
               
              4954
               
            </ListItem>
            <ListItem
              className="stat"
            >
               
              <span>
                views
                :
              </span>
               
              47030
               
            </ListItem>
          </List>
        </div>
      </div>
    </div>
    <div
      className="profile-preview"
    >
      <div
        className="profile-preview__panel profile-preview__panel--active"
        style={
          Object {
            "backgroundImage": "url(https://mir-s3-cdn-cf.behance.net/user/138/4609491853107.58b2df5a7fef4.jpg)",
            "backgroundPosition": "top right",
            "backgroundRepeat": "no-repeat",
          }
        }
      >
        <div
          className="profile-preview__header"
        >
          <div
            className="profile-nav"
          >
            <div>
              <h1>
                Following
              </h1>
              <List
                className="following"
              />
            </div>
            <div>
              <h1>
                Followers
              </h1>
              <List
                className="followers"
              />
            </div>
          </div>
        </div>
        <Divider />
        <div
          className="profile-preview__content work_experience"
        >
          <section>
            <h2>
              Work Experience
            </h2>
            <List />
          </section>
        </div>
      </div>
    </div>
  </div>
</div>
`;
